docs/source/dask-expr-api.rst

```diff
@@ -0,0 +1,603 @@
+Dask DataFrame API with Logical Query Planning
+==============================================
```

Comment:
It might be nice to add a few sentences here about the query planning effort and how to enable it. I don't mean this as a blocking comment -- we can always follow-up later, or not at all

Reply:
Yes I intend to add content as we go before the new release is out